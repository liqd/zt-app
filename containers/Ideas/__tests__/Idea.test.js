import React from 'react'
import { act, fireEvent, render, waitFor } from '@testing-library/react-native'

import '@testing-library/jest-native/extend-expect'

import API from '../../../BaseApi'
import { testComment as mockComment, testIdeaAdmin, testModule } from '../../../tests/TestData'
import { Idea } from '../Idea'

jest.mock('../../../BaseApi', () => ({
  __esModule: true,
  default: { getComments: jest.fn(() => Promise.resolve({ statusCode: 200, data: {results: [mockComment] }})) }
}))

test('Test Idea Snapshot', async () => {
  const route = { params: { idea: testIdeaAdmin, module: testModule } }
  const { toJSON, queryByText } = render(<Idea route={route} />)
  await waitFor(() => {
    expect(queryByText(/Lorem ipsum/)).toBeTruthy()
  })
  expect(toJSON()).toMatchSnapshot()
})

test('Test Idea Comment Menu No Permissions', async () => {
  const route = { params: { idea: testIdeaAdmin, module: testModule } }
  const { getByTestId, queryByText } = render(<Idea route={route} />)
  await waitFor(() => {
    expect(queryByText(/Lorem ipsum/)).toBeTruthy()
  })
  const menuButton = getByTestId('options_button_' + mockComment.id)
  fireEvent.press(menuButton)
  expect(queryByText('Edit')).toBeFalsy()
  expect(queryByText('Report')).toBeTruthy()
  expect(queryByText('Delete')).toBeFalsy()
})

test('Test Idea Comment Menu Edit Permissions', async () => {
  API.getComments = jest.fn(() => Promise.resolve({
    statusCode: 200,
    data: {
      results: [{
        ...mockComment,
        user_info: { ...mockComment.user_info, has_changing_permission: true }
      }]
    }
  }))
  const route = { params: { idea: testIdeaAdmin, module: testModule } }
  const { getByTestId, queryByText } = render(<Idea route={route} />)
  await waitFor(() => {
    expect(queryByText(/Lorem ipsum/)).toBeTruthy()
  })
  const menuButton = getByTestId('options_button_' + mockComment.id)
  fireEvent.press(menuButton)
  expect(queryByText('Edit')).toBeTruthy()
  expect(queryByText('Report')).toBeTruthy()
  expect(queryByText('Delete')).toBeFalsy()
})

test('Test Idea Comment Edit', async () => {
  API.getComments = jest.fn(() => Promise.resolve({
    statusCode: 200,
    data: {
      results: [{
        ...mockComment,
        user_info: { ...mockComment.user_info, has_changing_permission: true }
      }]
    }
  }))
  const route = { params: { idea: testIdeaAdmin, module: testModule } }
  const {
    findByText,
    getByPlaceholderText,
    getByTestId,
    getByText,
    queryByText
  } = render(<Idea route={route} />)
  // wait till hooks finished
  const comment = await findByText(/Lorem ipsum/)
  expect(comment).toBeTruthy()
  const commentFormInput = getByPlaceholderText('Enter your comment')
  // text input empty
  expect(commentFormInput).toHaveProp('value', '')
  const menuButton = getByTestId('options_button_' + mockComment.id)
  fireEvent.press(menuButton)
  // menu should be visible
  expect(queryByText('Edit')).toBeTruthy()
  const editButton = getByText('Edit')
  fireEvent.press(editButton)
  // menu should be gone
  expect(queryByText('Edit')).toBeFalsy()
  // text input filled with comment
  expect(commentFormInput).toHaveProp('value', mockComment.comment)
  await act(async () => {
    fireEvent.changeText(commentFormInput, 'This is an edited comment')
  })
  expect(commentFormInput).toHaveProp('value', 'This is an edited comment')
})
