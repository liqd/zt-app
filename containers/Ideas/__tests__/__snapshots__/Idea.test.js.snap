// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Idea Snapshot 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "paddingTop": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 20,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 2,
            },
            undefined,
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 2,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginVertical": 0,
                "padding": 0,
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 22,
                  },
                  undefined,
                  Object {
                    "fontFamily": "simple-line-icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#1a1a1a",
                  "fontSize": 17,
                  "marginHorizontal": 5,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Back
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 2,
            },
            undefined,
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 2,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 22,
                  },
                  undefined,
                  Object {
                    "fontFamily": "simple-line-icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      keyboardShouldPersistTaps="handled"
      onScroll={[Function]}
      scrollEnabled={true}
      style={
        Object {
          "backgroundColor": "#fff",
          "paddingHorizontal": 15,
        }
      }
    >
      <View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={Object {}}
        >
          <View
            style={
              Object {
                "marginTop": 5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "SourceSansPro_400Regular",
                  },
                  Object {
                    "fontSize": 17,
                    "fontWeight": "800",
                    "lineHeight": 20,
                  },
                ]
              }
            >
              App Test Idea 0
            </Text>
          </View>
          <View
            style={
              Object {
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "SourceSansPro_400Regular",
                  },
                  Object {
                    "fontSize": 15,
                  },
                ]
              }
            >
              This is a test idea.

  It's a really good one.
  
            </Text>
          </View>
          <View
            style={
              Object {
                "marginVertical": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "SourceSansPro_400Regular",
                  },
                  Object {
                    "fontSize": 15,
                    "lineHeight": 19,
                    "textTransform": "capitalize",
                  },
                ]
              }
            >
              admin
               
              June 29, 2022
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "SourceSansPro_400Regular",
                  },
                  Object {
                    "fontSize": 15,
                  },
                ]
              }
            >
              Reference No.: 
              2022-00006
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 25,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "width": 120,
                }
              }
            >
              <View
                accessibilityHint="click to up vote"
                accessibilityLabel="up-votes"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "SourceSansPro_400Regular",
                      },
                      Object {
                        "fontSize": 15,
                        "letterSpacing": 5,
                        "lineHeight": 19,
                      },
                    ]
                  }
                >
                  0
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "SourceSansPro_400Regular",
                      },
                      Object {
                        "fontSize": 15,
                        "letterSpacing": 5,
                        "lineHeight": 19,
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 18,
                        },
                        undefined,
                        Object {
                          "fontFamily": "simple-line-icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </Text>
              </View>
              <View
                accessibilityHint="click to down vote"
                accessibilityLabel="down-votes"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "SourceSansPro_400Regular",
                      },
                      Object {
                        "fontSize": 15,
                        "letterSpacing": 5,
                        "lineHeight": 19,
                      },
                    ]
                  }
                >
                  0
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "SourceSansPro_400Regular",
                      },
                      Object {
                        "fontSize": 15,
                        "letterSpacing": 5,
                        "lineHeight": 19,
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 18,
                        },
                        undefined,
                        Object {
                          "fontFamily": "simple-line-icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </Text>
              </View>
            </View>
            <View>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#1a1a1a",
                      "fontSize": 18,
                    },
                    Object {
                      "fontSize": 17,
                      "marginVertical": 10,
                    },
                    Object {
                      "fontFamily": "simple-line-icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View>
            <View>
              <Text
                style={
                  Object {
                    "fontFamily": "SourceSansPro_400Regular",
                  }
                }
              >
                1
                 
                Entry
              </Text>
              <View
                style={
                  Object {
                    "borderColor": "#c4c4c4",
                    "borderTopWidth": 1,
                    "marginVertical": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginVertical": 20,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Image
                      accessibilityIgnoresInvertColors={true}
                      source={
                        Object {
                          "uri": "",
                        }
                      }
                      style={
                        Object {
                          "backgroundColor": "#2d40cc",
                          "borderRadius": 20,
                          "height": 40,
                          "marginRight": 20,
                          "width": 40,
                        }
                      }
                    />
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontFamily": "SourceSansPro_400Regular",
                            },
                            Object {
                              "fontWeight": "800",
                            },
                          ]
                        }
                      >
                        test-user
                      </Text>
                      <Text
                        style={
                          Array [
                            Object {
                              "fontFamily": "SourceSansPro_400Regular",
                            },
                            Object {
                              "color": "#666",
                              "fontSize": 13,
                            },
                          ]
                        }
                      >
                        January 1, 2022
                      </Text>
                    </View>
                  </View>
                  <Text
                    style={
                      Object {
                        "fontFamily": "SourceSansPro_400Regular",
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 2,
                          },
                          undefined,
                          false,
                        ]
                      }
                      testID="RNE_BUTTON_WRAPPER"
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          Object {
                            "busy": false,
                            "disabled": false,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                        testID="options_button_1"
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "borderColor": "#2089dc",
                              "borderRadius": 2,
                              "borderWidth": 0,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "padding": 8,
                              "paddingHorizontal": 10,
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 22,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "simple-line-icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </Text>
                </View>
                <View>
                  <Text
                    numberOfLines={2}
                    onTextLayout={[Function]}
                    style={
                      Object {
                        "fontFamily": "SourceSansPro_400Regular",
                      }
                    }
                  >
                    Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginVertical": 20,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingBottom": 10,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "width": 120,
                      }
                    }
                  >
                    <View
                      accessibilityHint="click to up vote"
                      accessibilityLabel="up-votes"
                      accessibilityState={
                        Object {
                          "disabled": true,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontFamily": "SourceSansPro_400Regular",
                            },
                            Array [
                              Object {
                                "fontSize": 15,
                                "letterSpacing": 5,
                                "lineHeight": 19,
                              },
                              Object {
                                "color": "#c4c4c4",
                              },
                            ],
                          ]
                        }
                      >
                        0
                      </Text>
                      <Text
                        style={
                          Array [
                            Object {
                              "fontFamily": "SourceSansPro_400Regular",
                            },
                            Array [
                              Object {
                                "fontSize": 15,
                                "letterSpacing": 5,
                                "lineHeight": 19,
                              },
                              Object {
                                "color": "#c4c4c4",
                              },
                            ],
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 18,
                              },
                              undefined,
                              Object {
                                "fontFamily": "simple-line-icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          
                        </Text>
                      </Text>
                    </View>
                    <View
                      accessibilityHint="click to down vote"
                      accessibilityLabel="down-votes"
                      accessibilityState={
                        Object {
                          "disabled": true,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontFamily": "SourceSansPro_400Regular",
                            },
                            Array [
                              Object {
                                "fontSize": 15,
                                "letterSpacing": 5,
                                "lineHeight": 19,
                              },
                              Object {
                                "color": "#c4c4c4",
                              },
                            ],
                          ]
                        }
                      >
                        0
                      </Text>
                      <Text
                        style={
                          Array [
                            Object {
                              "fontFamily": "SourceSansPro_400Regular",
                            },
                            Array [
                              Object {
                                "fontSize": 15,
                                "letterSpacing": 5,
                                "lineHeight": 19,
                              },
                              Object {
                                "color": "#c4c4c4",
                              },
                            ],
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 18,
                              },
                              undefined,
                              Object {
                                "fontFamily": "simple-line-icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          
                        </Text>
                      </Text>
                    </View>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 2,
                        },
                        undefined,
                        false,
                      ]
                    }
                    testID="RNE_BUTTON_WRAPPER"
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        Object {
                          "busy": false,
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderColor": "#2089dc",
                            "borderRadius": 2,
                            "borderWidth": 0,
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "padding": 8,
                            "paddingHorizontal": 10,
                          }
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 18,
                              },
                              undefined,
                              Object {
                                "fontFamily": "simple-line-icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          
                        </Text>
                        <Text
                          style={
                            Object {
                              "color": "#464a4c",
                              "fontSize": 15,
                              "lineHeight": 19,
                              "marginLeft": 8,
                              "paddingVertical": 1,
                              "textAlign": "center",
                            }
                          }
                        >
                          Reply
                        </Text>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 2,
                        },
                        undefined,
                        false,
                      ]
                    }
                    testID="RNE_BUTTON_WRAPPER"
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        Object {
                          "busy": false,
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderColor": "#2089dc",
                            "borderRadius": 2,
                            "borderWidth": 0,
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "padding": 8,
                            "paddingHorizontal": 10,
                          }
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 18,
                              },
                              undefined,
                              Object {
                                "fontFamily": "simple-line-icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          
                        </Text>
                        <Text
                          style={
                            Object {
                              "color": "#464a4c",
                              "fontSize": 15,
                              "lineHeight": 19,
                              "marginLeft": 8,
                              "paddingVertical": 1,
                              "textAlign": "center",
                            }
                          }
                        >
                          Share
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 20,
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 18,
            "flex": 2,
            "flexDirection": "row",
            "padding": 10,
          }
        }
      >
        <TextInput
          accessibilityHint="Type your comment here"
          accessibilityLabel="Comment input"
          multiline={true}
          name="comment"
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Enter your comment"
          placeholderTextColor="#666"
          style={
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
              "width": "100%",
            }
          }
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#2d40cc",
            "borderRadius": 18,
            "marginLeft": 10,
            "padding": 5,
            "paddingRight": 8,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 17,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    />
  </View>
</RNCSafeAreaView>
`;
